packages/aws-cdk-lib/aws-cloudfront/test/function.test.ts

```diff
@@ -188,4 +188,89 @@ describe('CloudFront Function', () => {
       },
     });
   });
+
+  describe('key value store association', () => {
+    test('minimal example', () => {
+      const stack = new Stack();
+      const keyValueStore = new KeyValueStore(stack, 'TestStore');
+
+      new Function(stack, 'TestFn', {
+        code: FunctionCode.fromInline('code'),
+        keyValueStore,
+      });
+
+      Template.fromStack(stack).hasResourceProperties('AWS::CloudFront::Function', {
+        FunctionConfig: {
+          Runtime: 'cloudfront-js-2.0',
+          KeyValueStoreAssociations: [{
+            KeyValueStoreARN: stack.resolve(keyValueStore.keyValueStoreArn),
+          }],
+        },
+      });
+    });
+
+    test('rejects key value store with v1.0 runtime', () => {
+      const stack = new Stack();
+      const keyValueStore = new KeyValueStore(stack, 'TestStore');
+
+      expect(() => new Function(stack, 'TestFn', {
+        code: FunctionCode.fromInline('code'),
+        runtime: FunctionRuntime.JS_1_0,
+        keyValueStore,
+      })).toThrow(/Key Value Stores cannot be associated to functions using the .* runtime/);
+    });
+
+    test('defaults to js-2.0 runtime with key value store', () => {
+      const stack = new Stack();
+      const keyValueStore = new KeyValueStore(stack, 'TestStore');
+
+      new Function(stack, 'TestFn', {
+        code: FunctionCode.fromInline('code'),
+        runtime: undefined,
+        keyValueStore,
+      });
+
+      Template.fromStack(stack).hasResourceProperties('AWS::CloudFront::Function', {
+        FunctionConfig: {
+          Runtime: 'cloudfront-js-2.0',
+        },
+      });
+    });
+
+    test('works with js-2.0 runtime specified', () => {
+      const stack = new Stack();
+      const keyValueStore = new KeyValueStore(stack, 'TestStore');
+
+      new Function(stack, 'TestFn', {
+        code: FunctionCode.fromInline('code'),
+        runtime: FunctionRuntime.JS_2_0,
+        keyValueStore,
+      });
+
+      Template.fromStack(stack).hasResourceProperties('AWS::CloudFront::Function', {
+        FunctionConfig: {
+          Runtime: 'cloudfront-js-2.0',
+          KeyValueStoreAssociations: [{
+            KeyValueStoreARN: stack.resolve(keyValueStore.keyValueStoreArn),
+          }],
+        },
+      });
+    });
+
+    test('no value is used in CloudFormation when unspecified in CDK', () => {
+      const stack = new Stack();
+
+      new Function(stack, 'TestFn', {
+        code: FunctionCode.fromInline('code'),
+        runtime: FunctionRuntime.JS_2_0,
+        keyValueStore: undefined,
+      });
+
+      Template.fromStack(stack).hasResourceProperties('AWS::CloudFront::Function', {
+        FunctionConfig: {
+          KeyValueStoreAssociations: Match.absent(),
+        },
+      });
+    });
+  });
```

Comment:
I don't see much need for this, but what do you think? (unless you intend to make sure that it is not generated with an empty array, or something like that.)

```suggestion
```

Reply:
Yeah, I think I like it being explicit that it should be left absent instead of perhaps an empty list. I agree it's perhaps covered by other cases or less critical but it does feel like an important way to preserve the correct behavior at a CloudFormation layer.

Reply:
OK, I agree. Let's go ahead with this.